### PR TITLE
Fix for the Jira Issue : EMM-997 & EMM-1084

### PR DESCRIPTION
**Pull request by Kamidu**

### CODE REVIEW

File: modules/apps/jaggery/emm-web-agent/src/emm-web-agent/modules/backend-service-invoker.js [hunk 1]
```diff
@@ -63,18 +76,18 @@ var backendServiceInvoker = function () {
             } else if (xmlHttpRequest.status == 401 && (xmlHttpRequest.responseText == TOKEN_EXPIRED ||
                                                         xmlHttpRequest.responseText == TOKEN_INVALID ) && count < 5 ) {
                 tokenUtil.refreshToken();
-                return execute(count);
+                return execute(count+1);
```
Comment (kasundananjaya) [inline modules/apps/jaggery/emm-web-agent/src/emm-web-agent/modules/backend-service-invoker.js]: Correct the formatting issues.

File: modules/apps/jaggery/emm-web-agent/src/emm-web-agent/modules/backend-service-invoker.js [hunk 2]
```diff
@@ -63,18 +76,18 @@ var backendServiceInvoker = function () {
...
             }
         };
-        var accessToken = session.get(constants.ACCESS_TOKEN_PAIR_IDENTIFIER).accessToken.trim();
+        var accessToken = getAccessToken();
         if (accessToken) {
-            return execute(1);
+            return execute(0);
         } else {
-           response.status = 401;
+            response.sendRedirect( mdmProps["httpsURL"]+"/emm/login");
```
Comment (kasundananjaya) [inline modules/apps/jaggery/emm-web-agent/src/emm-web-agent/modules/backend-service-invoker.js]: Fix formatting issues.

File: modules/apps/jaggery/emm/src/emm/modules/backend-service-invoker.js [hunk 3]
```diff
@@ -29,7 +29,21 @@ var backendServiceInvoker = function () {
...
+    /**
+     * This methoad reads the token pair from the session and return the access token.
+     * If the token pair s not set in the session this will send a redirect to the login page.
+     */
+    function getAccessToken() {
+        var tokenPair = session.get(constants.ACCESS_TOKEN_PAIR_IDENTIFIER);
+        if (tokenPair) {
+            return tokenPair.accessToken ;
+        } else {
+            response.sendRedirect( mdmProps["httpsURL"]+"/emm/login");
```
Comment (kasundananjaya) [inline modules/apps/jaggery/emm/src/emm/modules/backend-service-invoker.js]: Fix formatting issues.

File: modules/apps/jaggery/emm/src/emm/modules/backend-service-invoker.js [hunk 4]
```diff
@@ -119,7 +132,7 @@ var backendServiceInvoker = function () {
         header.setValue(constants.APPLICATION_JSON);
         httpMethodObject.addRequestHeader(header);
```
Comment (harshanL) [inline modules/apps/jaggery/emm/src/emm/modules/backend-service-invoker.js]: What is the point of having !! ?

File: modules/apps/jaggery/emm/src/emm/units/appbar/public/js/appbar.js [hunk 5]
```diff
@@ -77,6 +77,9 @@ function loadNotifications(){
     var serviceURL = "/mdm-admin/notifications/NEW";
 
     var successCallback = function (data) {
+        if(data){
```
Comment (harshanL) [inline modules/apps/jaggery/emm/src/emm/units/appbar/public/js/appbar.js]: Fix formatting of if blocks here.

File: modules/apps/jaggery/emm-web-agent/src/emm-web-agent/modules/backend-service-invoker.js (hunk 1 above)
Comment (Kamidu) [inline modules/apps/jaggery/emm-web-agent/src/emm-web-agent/modules/backend-service-invoker.js]: These formatting issues has been fixed in a recent PR

File: modules/apps/jaggery/emm/src/emm/units/appbar/public/js/appbar.js (hunk 5 above)
Comment (Kamidu) [inline modules/apps/jaggery/emm/src/emm/units/appbar/public/js/appbar.js]: These formatting issues has been fixed in a recent PR

File: modules/apps/jaggery/emm/src/emm/modules/backend-service-invoker.js (hunk 4 above)
Comment (Kamidu) [inline modules/apps/jaggery/emm/src/emm/modules/backend-service-invoker.js]: This is the content type header for the http request.there is a design process going to enable the content type and customer header to the invoker. That might change the source but the purpose of this is to indicates the media type of the entity-body sent to the recipient

**Pull request merged**